File: server/src/main/scala/producer/pulsar_producer_config/batcher_builder/KeyBasedBatcherBuilder.scala
package producer.pulsar_producer_config.batcher_builder

import com.tools.teal.pulsar.ui.producer.v1.producer as pb

case class KeyBasedBatcherBuilder()

object KeyBasedBatcherBuilder:
    def fromPb(v: pb.KeyBasedBatcherBuilder): KeyBasedBatcherBuilder =
        KeyBasedBatcherBuilder()

    def toPb(v: KeyBasedBatcherBuilder): pb.KeyBasedBatcherBuilder =
        pb.KeyBasedBatcherBuilder()

File: ui/components/ui/ProducerSession/pulsar-producer-config/batcher-builder/key-based-batcher-builder.ts
import * as pb from '../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';

export type KeyBasedBatcherBuilder = {
  type: 'key-based-batcher-builder'
};

export function keyBasedBatcherBuilderFromPb(v: pb.KeyBasedBatcherBuilder): KeyBasedBatcherBuilder {
  return { type: 'key-based-batcher-builder' };
}

export function keyBasedBatcherBuilderToPb(v: KeyBasedBatcherBuilder): pb.KeyBasedBatcherBuilder {
  return new pb.KeyBasedBatcherBuilder();
}
